pyhf/pdf.py

```diff
@@ -97,13 +106,13 @@ def add_or_get_modifier(self, channel, sample, modifier_def):
 
 class Model(object):
     def __init__(self, spec, **config_kwargs):
-        self.spec = spec
+        self.spec = copy.deepcopy(spec) #may get modified by config
         self.schema = config_kwargs.get('schema', utils.get_default_schema())
         # run jsonschema validation of input specification against the (provided) schema
         log.info("Validating spec against schema: {0:s}".format(self.schema))
         utils.validate(self.spec, self.schema)
         # build up our representation of the specification
-        self.config = _ModelConfig.from_spec(spec,**config_kwargs)
+        self.config = _ModelConfig.from_spec(self.spec,**config_kwargs)
```

Comment:
I prefer that the `copy.deepcopy()` is on this line. Throw away the `deepcopy`.

Reply:
not sure I follow, do you want the deepcopy removed or moved here?